src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -428,31 +453,25 @@ protected override Task AppendProjectStateDiagnosticsAsync(Project project, Docu
 
             private async Task<CompilationWithAnalyzers> GetCompilationWithAnalyzersAsync(Project project, bool concurrentAnalysis, bool reportSuppressedDiagnostics, CancellationToken cancellationToken)
             {
-                CompilationWithAnalyzers compilationWithAnalyzers;
-                lock (_compilationWithAnalyzersMap)
-                {
-                    if (_compilationWithAnalyzersMap.TryGetValue(project.Id, out compilationWithAnalyzers))
-                    {
-                        return compilationWithAnalyzers;
-                    }
-                }
-
-                if (project.SupportsCompilation)
+                if (_compilationWithAnalyzers == null && project.SupportsCompilation)
```

Comment:
hmmm.. all entry points for this create new IDELatestDiagnosticGetter that are tied to one specific project, so unless there is other code path for IDELatestDiagnosticGetter there is no way that map get shared between 2 projects. and IDELatestDiagnosticGetter is the one implementation of LatestDiagnosticGetter (I think there were more than one before, but it looks like now we have only one not sure what happened to other one)